pygmt/clib/session.py

```diff
@@ -2311,3 +2312,35 @@ def extract_region(self) -> np.ndarray:
         if status != 0:
             raise GMTCLibError("Failed to extract region from current figure.")
         return region
+
+
+def _raster_kind(raster: str) -> Literal["grid", "image"]:
+    """
+    Determine the raster kind.
+
+    Examples
+    --------
+    >>> _raster_kind("@earth_relief_01d")
+    'grid'
+    >>> _raster_kind("@static_earth_relief.nc")
+    'grid'
+    >>> _raster_kind("@earth_day_01d")
+    'image'
+    >>> _raster_kind("@hotspots.txt")
+    'grid'
+    """
+    # The logic here is because: an image can be read into a grid container, but a grid
+    # can't be read into an image container. So, try to read the file as an image first.
+    # If fails, try to read it as a grid.
```

Comment:
It turns out the logic here is not 100% correct.

- An image can be read into a grid container, but only the first band is read.
- A grid can be read into an image container, as long as `libgdal-netcdf` or `libgdal-hdf5` is installed

Reply:
> * A grid can be read into an image container, as long as another libgdal package (maybe `libgdal-netcdf` or `libgdal-hdf5`) is installed. xref:

The first try-except check is to try reading into a `GMT_IMAGE` struct, and return "image" if there are 3 bands, and "grid" if there is only 1 band. Even if grids can be read into `GMT_IMAGE`, we will still call it a grid if it has 1-band right?

Reply:
Trying to refresh my memory about what's happening:

- Reading a grid into `GMT_GRID`: Works!
- Reading an image into `GMT_GRID`: Only the first band is read.
- Reading a grid into `GMT_IMAGE`: It depends on `libgdal-hdf5` being installed. Read as an single-band image if installed, otherwise raise errors.
- Reading an image into `GMT_IMAGE`: works.


```
import pygmt
pygmt.grdcut("@earth_relief_01d_p", region=[0, 10, 0, 10])
```
If `libgdal-hdf5` is not installed, `_raster_kind` first read it as an image, which fails and reports following errors:
```
ERROR 4: `/home/runner/.gmt/server/earth/earth_relief/earth_relief_01d_p.grd' not recognized as being in a supported file format. It could have been recognized by driver HDF5, but plugin gdal_HDF5.so is not available in your installation. You may install it with 'conda install -c conda-forge libgdal-hdf5'
Error: ession [ERROR]: Unable to open earth_relief_01d_p.grd.
Error: ession [ERROR]: ERROR reading image with gdalread.
pygmt-session (gmtapi_import_image): Could not read from file [earth_relief_01d_p.grd]
[Session pygmt-session (43)]: Error returned from GMT API: GMT_IMAGE_READ_ERROR (22)
[Session pygmt-session (43)]: Error returned from GMT API: GMT_IMAGE_READ_ERROR (22)
```
then it tries to read it as a grid, which succeeds. But we still see the error messages above. I believe we need to find a way to suppress these errors or find an alternative way to determine the data kind.